### PR TITLE
fix: increase save button hitslop

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Bug that came out of Mobile QA, save button not easily pressable.

- [x] Makes hitslop of save button bigger to be easier to press
- [x] adds customHitslop to palette button
- [x] bell icon wasn't working properly either when on dark button -> wasn't visible so fixed that as well
- [x] placeholder adjustment 

#### Screenshots (with touchable highlighted from RN inspector)

|Before|After|
|---|---|
|IMG_8685|IMG_8684|

|More screenshots|||
|---|---|---|
|IMG_1C12F1291251-2|IMG_1C12F1291251-1|IMG_25A02E0D4E1B-1|

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- increase save button hitslop - gkartalis

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by dimatretyak — COMMENTED** [review]
Unfortunately, the problem is still reproducible on my device. I'll take a look at the problem on Monday and try to fix it

https://user-images.githubusercontent.com/3513494/219748265-79d97cc7-5c7c-4729-bfec-9bd90b9bbf7e.mp4

**Comment by gkartalis** [comment]
Found the reason for this, Pressable inside button does not inherit the rest of the props thus hitslop is not being used anywhere and that's why it is not changing the behavior at all.

Will push a fix here

**Comment by gkartalis** [comment]
@pvinis let me know your thoughts, would we also want to support something like this on external palette as well? 

the text variant button wasn't that easy to press, I guess due to not visible boundaries like the rest of the buttons

File: src/palette/Button/Button.tsx [hunk 1]
```diff
@@ -53,6 +53,9 @@ export interface ButtonProps extends BoxProps {
   testOnly_state?: DisplayState
 
   textVariant?: TextVariant
+
+  /** customHitslop: used only if you want the button to have bigger hitslop than its actual layout */
+  customHitslop?: ButtonProps["hitSlop"]
```
Comment (dimatretyak) [inline src/palette/Button/Button.tsx:58]: **question**: can we rename `customHitslop` to `hitslop`? Sorry if the question is stupid 😅

**Review by dimatretyak — APPROVED** [review]

File: src/palette/Button/Button.tsx (hunk 1 above)
Comment (lordkiz) [reply src/palette/Button/Button.tsx:58]: Yeah, I also think this is better named hitslop

Comment (gkartalis) [reply src/palette/Button/Button.tsx:58]: also agree 👍 on it

Comment (dimatretyak) [reply src/palette/Button/Button.tsx:58]: thank you!

File: src/palette/Button/Button.tsx [hunk 2]
```diff
@@ -141,6 +141,7 @@ export const Button: React.FC<ButtonProps> = ({
     <Spring native to={to} config={config.stiff}>
       {(springProps: typeof to) => (
         <Pressable
+          hitSlop={rest?.hitSlop}
```
Comment (pvinis) [inline src/palette/Button/Button.tsx:144]: id suggest you get `hitSlop` from the props, and just use it without getting it from rest. put it directly on line 77 or something.

**Review by pvinis — DISMISSED** [review]
nice thats a good change. we def have this manually added on folio, so i think adding this to palette-mobile makes sense. maybe even make it default? or use a prop like `bigHitSlop` that adds these?

**Comment by pvinis** [comment]
(your example image reminds me of )

**Review by lordkiz — APPROVED** [review]

**Comment by gkartalis** [comment]
also fixed something extra, placeholder adjustment and bell icon background

**Pull request merged**